Makefile

```diff
@@ -106,3 +106,7 @@ cluster-sync-importer: WHAT = cmd/cdi-importer
 cluster-sync-importer: cluster-sync
 cluster-sync-cloner: WHAT = cmd/cdi-cloner
 cluster-sync-cloner: cluster-sync
+
+functest:
```

Comment:
Today we have `test-functional` as a make target. We need to disambiguate the term _functional_. CDI originally used _functional_ to mean multi-level mocked tests -- more system-level than unit tests but no running cluster required. CDI considered e2e tests to be similar to kubernete's and openshift's definition of e2e where system-level tests are run on a live cluster. However, kubevirt has a different definition of "functional". Our _Makefile_ should not have multiple targets with the name "func" or "functional" that have  different meanings, IMO.

Reply:
So do we want to rename mine to e2e or something or do we want to rename test-functional, to test-unit-extended or something? I am fine with either one.

Reply:
I think I would prefer to stay consistent with kubevirt, but I can be persuaded either way

Reply:
I tend to agree that we should track kubevirt (even where it deviates from kubernetes). So, we need to figure out what to do with the existing cdi functional-tests (which are closer to unit tests than e2e). @copejon @screeley44 @davidvossel @aglitke any suggestions?

Reply:
IMO we need to parallel kubevirt's nomenclature.  Given that, tests in test/functional should be moved into the directories of the components they exercise.  This refactor would be relatively simple as the build flow is generic wrt to packages.

This means that, post refactor, `make test-unit` would execute all our existing tests and `make test-functional` would execute any functional (in the kubevirt definition) under test/functional

Reply:
Do we tackle that in this PR or make a new one?

Reply:
I think it would be better to do that in a separate PR.

Reply:
so based on @copejon's comment above, @awels it's ok to keep your make target names in tact, etc, and the new pr will address moving existing functional tests into pkg dirs, and changing your functional test target names if desired.